mobile/app/(app)/change-password.tsx

```diff
@@ -0,0 +1,189 @@
+import { YStack } from "tamagui";
+import Card from "../../components/Card";
+import Button from "../../components/Button";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+import { Control, Controller, useForm } from "react-hook-form";
+import React from "react";
+import { Typography } from "../../components/Typography";
+import FormInput from "../../components/FormInputs/FormInput";
+import { useTranslation } from "react-i18next";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import ChangePasswordConfirmation from "./change-password-confirmation";
+import { useChangePasswordMutation } from "../../services/mutations/change-password.mutation";
+import { ChangePasswordPayload } from "../../services/definitions.api";
+
+interface FormData {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+}
+
+const ChangePassowrd = () => {
+  const insets = useSafeAreaInsets();
+  const { t } = useTranslation("change_password");
+
+  // Form validation schema
+  const formSchema = z
+    .object({
+      currentPassword: z.string().min(1, t("form.current_password.required")),
+      newPassword: z
+        .string()
+        .min(8, t("form.new_password.helper"))
+        .regex(/^(?=.*[A-Z])(?=.*[0-9])(?=.*[^a-zA-Z0-9]).*$/, t("form.new_password.pattern")),
+      confirmPassword: z.string(),
+    })
+    .refine((data) => data.newPassword === data.confirmPassword, {
+      message: t("form.confirm_password.no_match"),
+      path: ["confirmPassword"],
+    });
+
+  // React Hook Form setup
+  const { handleSubmit, control, formState } = useForm<FormData>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      currentPassword: "Votemonitor1*",
+      newPassword: "Votemonitor1*",
+      confirmPassword: "Votemonitor1*",
+    },
+  });
+  const { errors } = formState;
+
+  // Submit handler - change password
+  const {
+    mutate: updatePassword,
+    isError: credentialsError,
+    isSuccess: succesfullyChanged,
+  } = useChangePasswordMutation();
+
+  const onSubmit = async (data: FormData) => {
+    const payload: ChangePasswordPayload = {
+      password: data.currentPassword.trim(),
+      newPassword: data.newPassword.trim(),
+      confirmNewPassword: data.confirmPassword.trim(),
+    };
+    updatePassword(payload);
+  };
+
+  // Render either form or confirmation screen
+  return succesfullyChanged ? (
+    <ChangePasswordConfirmation />
+  ) : (
+    <Screen
+      preset="auto"
+      backgroundColor="white"
+      ScrollViewProps={{
+        bounces: false,
+      }}
+      contentContainerStyle={$containerStyle}
+    >
+      <Header
+        title={t("header.title")}
+        titleColor="white"
+        barStyle="light-content"
+        leftIcon={<Icon icon="chevronLeft" color="white" />}
+        onLeftPress={() => router.back()}
+      />
+
+      <YStack gap="$xl" paddingHorizontal={16} paddingVertical={40}>
```

Comment:
the space between the 3 password inputs doesn't seem to be the same (the ones that have helper text are closer, probably because you reserve space for helper text even if you don't have any)

Reply:
Done.
It was caused because I displayed and empty helper line for the current password.